plone/app/z3cform/widgets/base.py

```diff
@@ -89,7 +89,7 @@ def attributes(self):
 
         attributes = super().attributes
 
-        if self.field.required:
+        if self.required:
```

Comment:
I tried to follow back where `self.required` might be set and found it in `z3c.form.widget.Widget`.
So :+1: